Connection refused in PgConnectionAutoRetryTest

### Version

Which version(s) did you encounter this bug? 4.0.0, 4.0.1-SNAPSHOT cc3b6c99c7110ad98fd350c3d7f51b997c907305

### Context

Trying to build vertx-pg-client fails in these tests:
PgConnectionAutoRetryTest.testConnRetrySuccess
PgConnectionAutoRetryTest.testConnSuccessWithoutRetry
PgConnectionAutoRetryTest.testPoolRetrySuccess
PgConnectionAutoRetryTest.testPoolSuccessWithoutRetry

```
testConnSuccessWithoutRetry(io.vertx.pgclient.tck.PgConnectionAutoRetryTest)  Time elapsed: 1.022 sec  <<< ERROR!
io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /172.17.0.1:45965
Caused by: java.net.ConnectException: Connection refused
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:714)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:330)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:334)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:702)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:576)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
```

The Exception is reported in
https://github.com/eclipse-vertx/vertx-sql-client/blob/4.0.0/vertx-pg-client/src/test/java/io/vertx/pgclient/tck/ClientConfig.java#L43
https://github.com/eclipse-vertx/vertx-sql-client/blob/4.0.0/vertx-pg-client/src/test/java/io/vertx/pgclient/tck/ClientConfig.java#L65

### Do you have a reproducer?

https://github.com/eclipse-vertx/vertx-sql-client/blob/master/vertx-pg-client/src/test/java/io/vertx/pgclient/tck/PgConnectionAutoRetryTest.java

### Steps to reproduce

Build on Jenkins:
https://jenkins-aws.indexdata.com/job/folio-org/job/vertx-sql-client/job/master-vertx/
Complete log is available in the red box in the "Maven build" column.

--- a/vertx-db2-client/src/test/java/io/vertx/db2client/tck/DB2ConnectionAutoRetryTest.java
+++ b/vertx-db2-client/src/test/java/io/vertx/db2client/tck/DB2ConnectionAutoRetryTest.java
@@ -42,6 +42,7 @@ public class DB2ConnectionAutoRetryTest extends ConnectionAutoRetryTestBase {
   protected void initialConnector(int proxyPort) {
     SqlConnectOptions proxyOptions = new DB2ConnectOptions(options);
     proxyOptions.setPort(proxyPort);
+    proxyOptions.setHost("localhost");
     connectionConnector = ClientConfig.CONNECT.connect(vertx, proxyOptions);
     poolConnector = ClientConfig.POOLED.connect(vertx, proxyOptions);
   }

--- a/vertx-mssql-client/src/test/java/io/vertx/mssqlclient/tck/MSSQLConnectionAutoRetryTest.java
+++ b/vertx-mssql-client/src/test/java/io/vertx/mssqlclient/tck/MSSQLConnectionAutoRetryTest.java
@@ -42,6 +42,7 @@ public class MSSQLConnectionAutoRetryTest extends ConnectionAutoRetryTestBase {
   protected void initialConnector(int proxyPort) {
     SqlConnectOptions proxyOptions = new MSSQLConnectOptions(options);
     proxyOptions.setPort(proxyPort);
+    proxyOptions.setHost("localhost");
     connectionConnector = ClientConfig.CONNECT.connect(vertx, proxyOptions);
     poolConnector = ClientConfig.POOLED.connect(vertx, proxyOptions);
   }

--- a/vertx-mysql-client/src/test/java/io/vertx/mysqlclient/tck/MySQLConnectionAutoRetryTest.java
+++ b/vertx-mysql-client/src/test/java/io/vertx/mysqlclient/tck/MySQLConnectionAutoRetryTest.java
@@ -42,6 +42,7 @@ public class MySQLConnectionAutoRetryTest extends ConnectionAutoRetryTestBase {
   protected void initialConnector(int proxyPort) {
     SqlConnectOptions proxyOptions = new MySQLConnectOptions(options);
     proxyOptions.setPort(proxyPort);
+    proxyOptions.setHost("localhost");
     connectionConnector = ClientConfig.CONNECT.connect(vertx, proxyOptions);
     poolConnector = ClientConfig.POOLED.connect(vertx, proxyOptions);
   }

--- a/vertx-pg-client/src/test/java/io/vertx/pgclient/tck/PgConnectionAutoRetryTest.java
+++ b/vertx-pg-client/src/test/java/io/vertx/pgclient/tck/PgConnectionAutoRetryTest.java
@@ -42,6 +42,7 @@ public class PgConnectionAutoRetryTest extends ConnectionAutoRetryTestBase {
   protected void initialConnector(int proxyPort) {
     SqlConnectOptions proxyOptions = new PgConnectOptions(options);
     proxyOptions.setPort(proxyPort);
+    proxyOptions.setHost("localhost");
     connectionConnector = ClientConfig.CONNECT.connect(vertx, proxyOptions);
     poolConnector = ClientConfig.POOLED.connect(vertx, proxyOptions);
   }

--- a/vertx-sql-client/src/test/java/io/vertx/sqlclient/tck/ConnectionAutoRetryTestBase.java
+++ b/vertx-sql-client/src/test/java/io/vertx/sqlclient/tck/ConnectionAutoRetryTestBase.java
@@ -59,7 +59,7 @@ public abstract class ConnectionAutoRetryTestBase {
     options.setReconnectAttempts(3);
     options.setReconnectInterval(1000);
     UnstableProxyServer unstableProxyServer = new UnstableProxyServer(0);
-    unstableProxyServer.initialize(options.getPort(), ctx.asyncAssertSuccess(v -> {
+    unstableProxyServer.initialize(options, ctx.asyncAssertSuccess(v -> {
       initialConnector(unstableProxyServer.port());
       connectionConnector.connect(ctx.asyncAssertSuccess(conn -> {
         conn.close();
@@ -73,7 +73,7 @@ public abstract class ConnectionAutoRetryTestBase {
     options.setReconnectAttempts(3);
     options.setReconnectInterval(1000);
     UnstableProxyServer unstableProxyServer = new UnstableProxyServer(0);
-    unstableProxyServer.initialize(options.getPort(), ctx.asyncAssertSuccess(v -> {
+    unstableProxyServer.initialize(options, ctx.asyncAssertSuccess(v -> {
       initialConnector(unstableProxyServer.port());
       poolConnector.connect(ctx.asyncAssertSuccess(conn -> {
         conn.close();
@@ -86,7 +86,7 @@ public abstract class ConnectionAutoRetryTestBase {
     options.setReconnectAttempts(1);
     options.setReconnectInterval(1000);
     UnstableProxyServer unstableProxyServer = new UnstableProxyServer(2);
-    unstableProxyServer.initialize(options.getPort(), ctx.asyncAssertSuccess(v -> {
+    unstableProxyServer.initialize(options, ctx.asyncAssertSuccess(v -> {
       initialConnector(unstableProxyServer.port());
       connectionConnector.connect(ctx.asyncAssertFailure(throwable -> {
       }));
@@ -98,7 +98,7 @@ public abstract class ConnectionAutoRetryTestBase {
     options.setReconnectAttempts(1);
     options.setReconnectInterval(1000);
     UnstableProxyServer unstableProxyServer = new UnstableProxyServer(2);
-    unstableProxyServer.initialize(options.getPort(), ctx.asyncAssertSuccess(v -> {
+    unstableProxyServer.initialize(options, ctx.asyncAssertSuccess(v -> {
       initialConnector(unstableProxyServer.port());
       poolConnector.connect(ctx.asyncAssertFailure(throwable -> {
       }));
@@ -110,7 +110,7 @@ public abstract class ConnectionAutoRetryTestBase {
     options.setReconnectAttempts(1);
     options.setReconnectInterval(1000);
     UnstableProxyServer unstableProxyServer = new UnstableProxyServer(1);
-    unstableProxyServer.initialize(options.getPort(), ctx.asyncAssertSuccess(v -> {
+    unstableProxyServer.initialize(options, ctx.asyncAssertSuccess(v -> {
       initialConnector(unstableProxyServer.port());
       connectionConnector.connect(ctx.asyncAssertSuccess(connection -> {
         connection.close();
@@ -123,7 +123,7 @@ public abstract class ConnectionAutoRetryTestBase {
     options.setReconnectAttempts(1);
     options.setReconnectInterval(1000);
     UnstableProxyServer unstableProxyServer = new UnstableProxyServer(1);
-    unstableProxyServer.initialize(options.getPort(), ctx.asyncAssertSuccess(v -> {
+    unstableProxyServer.initialize(options, ctx.asyncAssertSuccess(v -> {
       initialConnector(unstableProxyServer.port());
       poolConnector.connect(ctx.asyncAssertSuccess(conn -> {
         conn.close();
@@ -149,7 +149,7 @@ public abstract class ConnectionAutoRetryTestBase {
       this.counter = new AtomicInteger(retryTimes);
     }
 
-    public void initialize(int targetPort, Handler<AsyncResult<Void>> resultHandler) {
+    public void initialize(SqlConnectOptions targetOptions, Handler<AsyncResult<Void>> resultHandler) {
       this.netClient = vertx.createNetClient();
       this.netServer = vertx.createNetServer()
         .connectHandler(frontendSocket -> {
@@ -171,7 +171,7 @@ public abstract class ConnectionAutoRetryTestBase {
             frontendSocket.close();
           } else {
             // pipe the stream to the database otherwise
-            netClient.connect(targetPort, "localhost")
+            netClient.connect(targetOptions.getPort(), targetOptions.getHost())
               .onSuccess(backendSocket -> {
                 LOGGER.info("Proxy: backend socket connected");
                 frontendSocketToBackendSocket.put(frontendSocket, backendSocket);

